src/components/renderer/SvgRenderer.tsx

```diff
@@ -0,0 +1,189 @@
+import { OpenAttestationDocument, v2, v4, utils } from "@govtechsg/open-attestation";
+import React, { CSSProperties, FunctionComponent, useEffect, useState } from "react";
+import { renderToStaticMarkup } from "react-dom/server";
+import { TextEncoder, TextDecoder } from "util";
+import crypto from "crypto";
+import bs58 from "bs58";
+import { ConnectionFailureTemplate, NoTemplate, SvgModifiedTemplate } from "../../DefaultTemplate";
+/* eslint-disable-next-line @typescript-eslint/no-var-requires */
+const handlebars = require("handlebars");
+
+interface SvgRendererProps {
+  document: OpenAttestationDocument;
+  svgRef: React.RefObject<HTMLIFrameElement>;
+  svgData?: string;
+  style?: CSSProperties;
+  className?: string;
+  sandbox?: string;
+  onConnected?: () => void; // Optional call method to call once svg is loaded
+  forceV2?: boolean;
+}
+
+const EMBEDDED_DOCUMENT = "[Embedded SVG]";
+
+enum DisplayResult {
+  OK = 0,
+  DEFAULT = 1,
+  CONNECTION_ERROR = 2,
+  DIGEST_ERROR = 3,
+}
+
+export const SvgRenderer: FunctionComponent<SvgRendererProps> = ({
+  document,
+  svgRef,
+  svgData,
+  style,
+  className,
+  sandbox = "allow-same-origin",
+  onConnected,
+  forceV2 = false,
+}) => {
+  const [buffer, setBuffer] = useState<ArrayBuffer>();
+  const [svgFetchedData, setFetchedSvgData] = useState<string>("");
+  const [source, setSource] = useState<string>("");
+  const [toDisplay, setToDisplay] = useState<DisplayResult>(DisplayResult.OK);
+
+  let docAsAny: any;
+  if (forceV2 && utils.isRawV2Document(docAsAny)) {
+    docAsAny = document as v2.OpenAttestationDocument;
+  } else {
+    docAsAny = document as any; // TODO: update type to v4.OpenAttestationDocument
+  }
```

Comment:
I'm not sure if we should utilise a `forceV2` flag? Because requires end-users manually detect and make sure they set the flag correctly.

Also, based on this logic, if someone puts in a v2 document but didn't set `forceV2 = true`, it will incorrectly assume it to be a v4 document?

Perhaps SVG rendering for v2 documents can still exist under `$template`:

```typescript
{
  // ...
  $template: {
    name: "SVG Certificate",
    type: "SvgRenderingTemplate2023", // or "SVG_RENDERER" to be similar to the current "EMBEDDED_RENDERER"
    url: "http://mockbucket.com/static/svg_test.svg",
    digestMultibase: "z97B2X5Qw9mVzyMhp46dQgWXtX7ipEALvg1coK26eS4f4",
  }
}
```

Ultimately, I think it's important to think about how we can generate the typings for v2 to support this SVG rendering without breaking change or introducing too big of a difference.

Reply:
I think it should be on the end user to decide and force this. 

On the logic, what do you think we should display instead? Once we update the typings a v2 doc should fail when being assumed as a v4 doc. But I am also open to the idea of displaying the wrongTemplate/noTemplate component instead.

I am against adding the SVG rendering under v2's `$template` as we are not officially supporting it for v2 and I want it to remain a "hacky" solution. If users want to do things right they should move to v4 before switching over to svgRendering.

Reply:
i also dont like this flag as its not very intuitive, and you are still kinda officially supporting v2

can we just officially type the `document` prop to `Pick<v4.OpenAttestationDocument, 'credentialSubject'|'renderMethod'>`

so in order for the user to use this with v2 document, they need to **explicitly(and hackily)** define v4's `credentialSubject` and `renderMethod`

Reply:
Create another parent adapter component to work with v2.

Reply:
Will change the document type to explicit properties and then add a v2 adapter to accept v2 docs and fit the v4 schema.